### PR TITLE
Updated assembly.ai integration

### PR DESCRIPTION
**Pull request by davidzhao**
Leveraging common utils including streaming resampler

### CODE REVIEW

**Review by theomonnom — APPROVED** [review]

**Comment by theomonnom** [comment]
Can you make the http_session optional;

File: livekit-plugins/livekit-plugins-assemblyai/livekit/plugins/assemblyai/stt.py [hunk 1]
```diff
@@ -66,9 +64,9 @@ def __init__(
         encoding: Optional[str] = "pcm_s16le",
         disable_partial_transcripts: bool = False,
         enable_extra_session_information: bool = False,
-        end_utterance_silence_threshold: int = 1000,
+        end_utterance_silence_threshold: Optional[int] = 100,
         http_session: Optional[aiohttp.ClientSession] = None,
-        buffer_size_seconds: float = 0.2,
+        buffer_size_seconds: float = 0.05,
```
Comment (theomonnom) [inline livekit-plugins/livekit-plugins-assemblyai/livekit/plugins/assemblyai/stt.py:69]: Not sure if this option is really relevant for the users. Let's pick the same one as DG?

Comment (davidzhao) [reply livekit-plugins/livekit-plugins-assemblyai/livekit/plugins/assemblyai/stt.py:69]: i didn't want to introduce additional latency, it seems that 50ms chunks is a good number for everything, wdyt?

Comment (theomonnom) [reply livekit-plugins/livekit-plugins-assemblyai/livekit/plugins/assemblyai/stt.py:69]: yeah 50ms seems good

Comment (theomonnom) [reply livekit-plugins/livekit-plugins-assemblyai/livekit/plugins/assemblyai/stt.py:69]: what I mean is that it is fine to hardcode it, but no big deal to leave it there

File: livekit-plugins/livekit-plugins-assemblyai/livekit/plugins/assemblyai/stt.py [hunk 2]
```diff
@@ -360,17 +320,17 @@ def _end_speech(self) -> None:
             ],
         )
         self._final_events = []
-        self._event_queue.put_nowait(end_event)
+        self._event_ch.send_nowait(end_event)
         # break async iteration if speech has ended
-        self._event_queue.put_nowait(None)
+        self._event_ch.send_nowait(None)
```
Comment (theomonnom) [inline livekit-plugins/livekit-plugins-assemblyai/livekit/plugins/assemblyai/stt.py:325]: We should never send None

Comment (theomonnom) [reply livekit-plugins/livekit-plugins-assemblyai/livekit/plugins/assemblyai/stt.py:325]: if the intent is to close, just do nothing, the STT baseclass will close the event_ch when the main task closes

**Pull request merged**